src/test/java/games/strategy/engine/framework/ArgParserTest.java

```diff
@@ -75,6 +78,16 @@ public void singleUrlArgIsAssumedToBeMapDownloadProperty() {
   }
 
   @Test
+  public void singleUrlArgIsUrlDecoded() throws UnsupportedEncodingException {
+    final String propvalue = "Something with spaces and Special chars 🤔";
+    final String testUrl = "triplea:" + URLEncoder.encode(propvalue, StandardCharsets.UTF_8.displayName());
```

Comment:
As I mentioned in a previous comment, `URLEncoder` encodes for the `application/x-www-form-urlencoded` media type instead of URI escaping per RFC 3986.  That means the encoded `testUrl` does not match what TripleA would actually receive from a browser.  The value of `testUrl` in the test is

```
triplea:Something+with+spaces+and+Special+chars+%F0%9F%A4%94
```

when it should be (generated by running the original string through JavaScript's `encodeURIComponent()` function)

```
triplea:Something%20with%20spaces%20and%20Special%20chars%20%F0%9F%A4%94
```

The code in `ArgParser` happens to work because `URLDecoder` still converts `+` to space.

My recommendation would be to avoid using `URLEncoder` entirely and simply URI escape the string by hand and hard-code it in the test.  That also makes it clearer to a test reader as to what's actually going on (and avoids all the noise caused by the `URLEncoder` charset requirement).  Something like:

```java
  @Test
  public void singleUrlArgIsUrlDecoded() {
    final String testUrl = "triplea:Something%20with%20spaces%20and%20Special%20chars%20%F0%9F%A4%94";
    ArgParser.handleCommandLineArgs(new String[] {testUrl}, new String[] {GameRunner.TRIPLEA_MAP_DOWNLOAD_PROPERTY});
    assertThat("if we pass only one arg prefixed with 'triplea:',"
        + " it should be properly URL-decoded as it's probably coming from a browser",
        System.getProperty(GameRunner.TRIPLEA_MAP_DOWNLOAD_PROPERTY), is("Something with spaces and Special chars 🤔"));
  }
```